stl/inc/chrono

```diff
@@ -6291,15 +6291,14 @@ namespace chrono {
             }
         }
 
-        _NODISCARD static array<_CharT, 4> _Fmt_string(const _Chrono_spec<_CharT>& _Spec) {
-            array<_CharT, 4> _Fmt_str;
+        _NODISCARD static basic_string<_CharT> _Fmt_string(const _Chrono_spec<_CharT>& _Spec) {
+            basic_string<_CharT> _Fmt_str(4, '\0');

```

Comment:
Using `basic_string` will degrade codegen. We should use an output buffer.

Reply:
How about tiny helper struct then?
```cpp
template <class _Ty, size_t _N>
struct _Fmt_string_data {
    _Ty _data[_N];
    constexpr _Ty* data() noexcept {
        return _data;
    }
};

_NODISCARD static _Fmt_string_data<_CharT, 4> _Fmt_string(const _Chrono_spec<_CharT>& _Spec) {
    _Fmt_string_data<_CharT, 4> _Fmt_str;
    size_t _Next_idx            = 0;
    _Fmt_str._data[_Next_idx++] = _CharT{'%'};
    if (_Spec._Modifier != '\0') {
        _Fmt_str._data[_Next_idx++] = static_cast<_CharT>(_Spec._Modifier);
    }
    _Fmt_str._data[_Next_idx++] = static_cast<_CharT>(_Spec._Type);
    _Fmt_str._data[_Next_idx]   = _CharT{'\0'};
    return _Fmt_str;
}
```

Reply:
I think output buffer would be simpler